### PR TITLE
Throw linting errors for using keywords and syntactic sugar

### LINKED ISSUE #194: throw error for reserved words
if I incorrectly use a reserved word in a DDL statement:
```sql
-- SQLite
CREATE TABLE artist(artist_name TEXT);  -- ok
CREATE TABLE artist(create TEXT);  -- `create` is a reserved word, syntax error
```

...should sqlfluff throw a lint error? Right now now it doesn't.

### PR DESCRIPTION
**Pull request by alanmcruickshank**
This resolves #194 .

Also adds a bit of syntactic sugar after #278 for managing keywords.

@Katzmann1983 - take a look at the first commit (before I make all the syntactic sugar changes) and you'll see how your work supported a rule which resolves the issue. Thanks for all your hard work to get us to this point! 👍

### CODE REVIEW

**Comment by Katzmann1983** [comment]
Do you think it would make sense to go even one step further and make the .keyword function the default function? I mean that
Ref('AnyThing') would automagically be converted by default into Ref.keyword('AnyThing'). Might make the code even more crispier...

File: src/sqlfluff/dialects/base.py [hunk 1]
```diff
@@ -39,10 +39,10 @@ def expand(self):
...
                 self.replace(**{key: self._library[key].expand(self)})
-        # Expand any sets as keywords:
-        for labeled_set in self._sets.values():  # e.g. reserved_keywords, (JOIN, ...)
+        # Expand any keyword sets.
+        for keyword_set in ['unreserved_keywords', 'reserved_keywords']:  # e.g. reserved_keywords, (JOIN, ...)
             # Make sure the values are available as KeywordSegments
-            for kw in labeled_set:
+            for kw in self.sets(keyword_set):
                 n = kw.capitalize() + 'KeywordSegment'
                 if n not in self._library:
```
[the remark this answers is not in the document]
Comment (alanmcruickshank) [reply src/sqlfluff/dialects/base.py:47]: I think a  way of having them synchronized makes a lot of sense. Probably not a replacement, but at least a constant.

**Comment by alanmcruickshank** [comment]
> the full way 'ANYTHING' -> Ref.keyword('ANYTHING')

I thought about this, and chickened out. I think you're right though. I'll have a go at going straight to this. It would make the code way more readable.

**Comment by alanmcruickshank** [comment]
@Katzmann1983 - see what you think of the most recent changes - we're now able to refer to keywords directly by name. The only weird case is the `ContainsOnly` grammar, which currently accepts strings already, but as a reference to types. At the moment I've not removed this functionality and just made that the only grammar to not accept strings as references to keywords (along with Ref itself).

I wondering whether to leave it as it, or whether to deprecate that usage of `ContainsOnly` and make it consistent with the rest.

**Pull request merged**